core/src/main/java/com/yahoo/ycsb/generator/Generator.java

```diff
@@ -18,20 +18,28 @@
 package com.yahoo.ycsb.generator;
 
 /**
- * An expression that generates a sequence of string values, following some distribution (Uniform, Zipfian, Sequential, etc.)
+ * An expression that generates a sequence of values, following some distribution (Uniform, Zipfian, Sequential, etc.)
  */
-public abstract class Generator 
+public abstract class Generator<V> 
 {
 	/**
-	 * Generate the next string in the distribution.
+	 * Generate the next value in the distribution.
 	 */
-	public abstract String nextString();
+	public abstract V nextValue();
 
 	/**
-	 * Return the previous string generated by the distribution; e.g., returned from the last nextString() call. 
-	 * Calling lastString() should not advance the distribution or have any side effects. If nextString() has not yet 
-	 * been called, lastString() should return something reasonable.
+	 * Return the previous value generated by the distribution; e.g., returned from the last {@link Generator#nextValue()} call. 
+	 * Calling {@link #lastValue()} should not advance the distribution or have any side effects. If {@link #nextValue()} has not yet 
+	 * been called, {@link #lastValue()} should return something reasonable.
 	 */
-	public abstract String lastString();
+	public abstract V lastValue();
+	
+	public final String nextString() {
+	  return nextValue().toString();
+	}
+	
+	public final String lastString() {
+	  return lastValue().toString();
+	}
```

Comment:
Yes, there's an implied contract here; numeric generators in existence will not return a `null` due to boxing of primitives. I will fix the string generators to comply with this assumption though.